### PR TITLE
fix/2829: force onChange to be updated with TinyMCE content before merge

### PR DESCRIPTION
**Pull request by BoardJames**
## Description
The issue #2829 was caused by the React state not correctly mirroring TinyMCE's state before the merge operation. This was despite the call to onChange directly before calling merge - however onChange was gated on TinyMCE's isDirty and it seems that the isDirty function had not updated to match reality in time (Note: I am told that isDirty functions are really hard to implement efficiently so, while we will tell the TinyMCE team about this, I doubt it is fixable within TinyMCE.). The easiest fix seems to be to force the update of React's state even if the editor doesn't think it has changed which is what I have done here.

## How Has This Been Tested?
Manually tested on Firefox and Chrome.

## Types of changes
Bug fix.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: blocks/editable/index.js [hunk 1]
```diff
@@ -287,8 +287,8 @@ export default class Editable extends Component {
 		}
 	}
 
-	onChange() {
-		if ( ! this.editor.isDirty() ) {
+	onChange( forced ) {
+		if ( ! forced && ! this.editor.isDirty() ) {
```
Comment (youknowriad) [inline blocks/editable/index.js]: Should we add a comment to explain that `isDirty` should be enough but due to a bug, we need to force sometimes?

**Review by youknowriad — APPROVED** [review]
This fixes the issue for me. A small comment might be good

**Review by mcsf — APPROVED** [review]
This fixes it for me. Minor suggestion: renaming `forced` to `shouldForce`, feel free to ignore.

File: blocks/editable/index.js (hunk 1 above)
Comment (EphoxJames) [reply blocks/editable/index.js]: On talking to the TinyMCE team it's more of a known deficiency needed due to efficiency requirements.

**Comment by EphoxJames** [comment]
Ok I will add a comment and rename forced to shouldForce.

**Pull request merged**